### ISSUE TITLE
Make the argument type of several extension functions xs:QName

### ISSUE THREAD

**Issue by ndw** [xproc/3.0-specification#687]
Several extension functions (`p:system-property`, `p:step-available`, ...) take a string as an argument and then have special rules about how it's transformed into a QName.

I propose that we make the arguments into `xs:QName`s and assert that the standard QName magic applies.

That'll work, right?

**Comment by xml-project** [comment]
Nope, because the rules we have so far are for variables/options with type `xs:QName` not for parameters. So we have we would have to add some more magic.
I would not resist this in general, but I think we should think about the consequences for XPath functions (standard and user-defined) in general. Treating "our" functions different from the other one seems odd, doesn't it.
